glean_parser/lint.py

```diff
@@ -286,6 +286,22 @@ def check_old_event_api(
         yield ("The old event API is gone. Extra keys require a type.")
 
 
+def check_metric_on_events_lifetime(
+    metric: metrics.Metric, parser_config: Dict[str, Any]
+) -> LintGenerator:
+    """A non-event metric on the events ping only makes sense if its lifetime
+    is tied to that events ping's lifetime."""
+    if (
+        "events" in metric.send_in_pings
+        and "all_pings" not in metric.send_in_pings
+        and metric.type != metrics.Event
+        and metric.lifetime != metrics.Lifetime.ping
+    ):
+        yield (
+            "Non-event metrics sent on the Events ping should have the ping lifetime"
```

Comment:
We usually refer to them by their lowercase name.
```suggestion
            "Non-event metrics sent on the events ping should have the ping lifetime"
```

Reply:
As said above, `ping` lifetime should be disallowed, `application` is what should be prefered.